src/torchmetrics/detection/mean_ap.py

```diff
@@ -579,10 +579,6 @@ def _evaluate_image(
         det = [det[i] for i in det_label_mask]
         if len(gt) == 0 and len(det) == 0:
             return None
-        if isinstance(det, dict):
-            det = [det]
-        if isinstance(gt, dict):
-            gt = [gt]
```

Comment:
`det` and `gt` are stored as `torch.Tensor`, not `dict`. Thus this checking is redundant.